ouroboros-consensus-test/src/Test/Util/Serialisation/Golden.hs

```diff
@@ -194,6 +194,9 @@ goldenTests testName examples enc goldenFolder
               Just label -> testName <> "_" <> label
       ]
   where
+    duplicateLabels :: [Maybe String]
+    duplicateLabels = labels \\ nub labels
```

Comment:
I suggest using `Data.Set` here.

- It's not quadratic.

- This is always a surprise: `"aabbcc" \\ "abc" = "abc"`. It's fine here but it's not ideal.